packages/pilot/src/components/withLoader/index.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { curry } from 'ramda'
+import { Transition, fade } from 'former-kit'
+
+import style from './style.css'
+
+const withRouter = curry((Renderer, Component) => {
+  const Loader = ({ loading, ...props }) => (
+    <div className={style.container}>
+      <Transition
+        atActive={fade.atActive}
+        atEnter={fade.atEnter}
+        atLeave={fade.atLeave}
+        mapStyles={fade.mapStyles}
+        springOptions={fade.springOptions}
+      >
+        {loading &&
+          React.cloneElement(Renderer, { key: Renderer.props.id || 'loader' })
```

Comment:
Esta key `loader` pode dar conflito com outros loaders quando o `Renderer` não fornecer id, sugiro um `shortid` para este queridão 😄 
```suggestion
          React.cloneElement(Renderer, { key: Renderer.props.id || `loader_${shortid.generate()}` })
```

Reply:
Uma dúvida sobre usar o shortID aqui é sobre re renderer, quando existir uma atualização ele vai mudar o valor do shotid certo? como fica isso, eu lembro que já discutimos sobre isso e ficou de usarmos sempre com um `state` mas não lembro exatamente do porque.

Reply:
Como o componente  é uma função não temos como garantir que o shortid rode só uma vez e vai sim redesenhar o elemento clonado, mas deste jeito ao menos garantimos que só este cara vai redesenhar, se houver conflito de keys mais componentes podem ser redesenhados devido ao conflito. Poderiamos evitar isso se o `WithLoader` fosse uma classe e o id criado no _constructor_ dela, mas não sei se seria a solução perfeita.

Reply:
Acho que faria sentido alterar ele pra ser uma classe então nesse caso.